x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/EngineIndexService.java

```diff
@@ -308,7 +303,47 @@ public void deleteEngine(String engineName, ActionListener<DeleteResponse> liste
         } catch (Exception e) {
             listener.onFailure(e);
         }
+    }
+
+    GetAliasesResponse getAlias(String engineAliasName) {
+        return clientWithOrigin.admin().indices().getAliases(new GetAliasesRequest(engineAliasName)).actionGet();
+    }
+
+    private void removeAlias(String engineAliasName, ActionListener<AcknowledgedResponse> listener) {
+        IndicesAliasesRequest aliasesRequest = new IndicesAliasesRequest().addAliasAction(
+            IndicesAliasesRequest.AliasActions.remove().aliases(engineAliasName).indices("*")
+        );
+        clientWithOrigin.admin().indices().aliases(aliasesRequest, ActionListener.wrap(response -> {
```

Comment:
Maybe listener.delegateFailure() would be an option as well to create the listener